pkg/util/oidcbuilder/oidcbuilder_test.go

```diff
@@ -0,0 +1,148 @@
+package oidcbuilder
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"reflect"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+
+	mock_azblob "github.com/Azure/ARO-RP/pkg/util/mocks/azblob"
+	utilerror "github.com/Azure/ARO-RP/test/util/error"
+)
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+func TestEnsureOIDCDocs(t *testing.T) {
```

Comment:
You could consider adding a few more test cases here to cover some other ways in which the code could fail:

- `Public key is not of type RSA` (the `default` case in the `switch`)
- `Failed to fetch key ID from public key`
- `JSON encoding of web key set failed` (if you can - I'm not sure if you can easily test this)

Reply:
Thanks!!
Added unit tests for the top 2 suggestions, but was not able to find any easy way to test the third `JSON encoding of web key set failed`.